src/core/MOM_density_integrals.F90

```diff
@@ -258,6 +280,10 @@ subroutine int_density_dz_generic_pcm(T, S, z_t, z_b, rho_ref, rho_0, G_e, HI, &
         hWght = max(0., -bathyT(i,j)-z_t(i+1,j), -bathyT(i+1,j)-z_t(i,j))
       if (top_massWeight) &
         hWght = max(hWght, z_b(i+1,j)-SSH(i,j), z_b(i,j)-SSH(i+1,j))
+      ! If both sides are nonvanished, then set it back to zero.
+      if (((z_t(i,j) - z_b(i,j)) > h_nonvanished) .and. ((z_t(i+1,j) - z_b(i+1,j)) > h_nonvanished)) then
+          hWght = massWeightNVonlyToggle * hWght
```

Comment:
Please reduce the indenting on lines 285 and 357 to follow MOM6's 2-point indenting convention.